custom/icds_reports/views.py

```diff
@@ -533,6 +533,10 @@ def get(self, request, *args, **kwargs):
         config.update(get_location_filter(location, domain))
         loc_level = get_location_level(config.get('aggregation_level'))
 
+        # Changing the aggregation level to the original convention which is 1 for state,
+        #  2 for district, 3 for block, 4 for sector and 5 for awc to optimise the db query
+        if config['aggregation_level'] > 1:
```

Comment:
setting the aggregation_level this way is fine for this case as we testing the speed improvement. But we need to make sure we do it in a reusable way if this works well and do it other places as well

Reply:
get_location_filter is aactually incrementing the aggregation value, we need to check and remove the addition if not necessary. Added a TODO